### PR TITLE
Task 5. Login check test

### PR DESCRIPTION
**Pull request by DianaKovall**
**FIFTH TASK — LOGIN CHECK(Google Fit app)**

 1. Login -> check if Home Page is opened  
 

~Login each time again

### CODE REVIEW

File: src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/AboutMePage.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+    @FindBy(id = "android:id/button1")
+    private ExtendedWebElement okayButton;
+
+    @FindBy(id = "com.google.android.apps.fitness:id/%s_field")
+    private ExtendedWebElement itemByID;
+
+    @FindBy(xpath = "//*[@class = 'android.widget.TextView'][1]")
+    private ExtendedWebElement title;
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/AboutMePage.java:27]: find better locator for title

File: src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/AboutMePage.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public TrackActivitiesPageBase clickNextBtn() {
+        nextButton.click();
+        return initPage(getDriver(), TrackActivitiesPageBase.class);
+    }
+
+
+    @Override
+    public AboutMePageBase clickOnBtn(PersonCharacteristics personCharacteristics) {
+        itemByID.format(personCharacteristics.getCharacteristic()).click();
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/AboutMePage.java:53]: have many pages can use the same id in PersonCharacteristics?

File: src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/AboutMePage.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import koval.myfit.mobile.gui.pages.common.loginPages.AboutMePageBase;
+import koval.myfit.mobile.gui.pages.common.loginPages.TrackActivitiesPageBase;
+import koval.myfit.mobile.gui.pages.service.enums.LoginPagesTitles;
+import koval.myfit.mobile.gui.pages.service.enums.PersonCharacteristics;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = AboutMePageBase.class)
+public class AboutMePage extends AboutMePageBase implements IMobileUtils, IConstantUtils {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/AboutMePage.java]: Please create AbstractPage with implements IMobileUtils, IConstantUtils. and extends it each PageBase. 
you will avoid duplicating (implements IMobileUtils, IConstantUtils)

File: src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/TrackActivitiesPage.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import koval.myfit.mobile.gui.pages.common.loginPages.TrackActivitiesPageBase;
+import koval.myfit.mobile.gui.pages.common.menuPages.HomePageBase;
+import koval.myfit.mobile.gui.pages.service.enums.LoginPagesTitles;
+import koval.myfit.mobile.gui.pages.service.enums.TrackYourActivitiesAnswer;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = TrackActivitiesPageBase.class)
+public class TrackActivitiesPage extends TrackActivitiesPageBase implements IMobileUtils, IConstantUtils {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/TrackActivitiesPage.java]: see above

File: src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/WelcomePage.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import koval.carinademo.mobile.gui.pages.service.interfaces.IConstantUtils;
+import koval.myfit.mobile.gui.pages.common.loginPages.AboutMePageBase;
+import koval.myfit.mobile.gui.pages.common.loginPages.WelcomePageBase;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = WelcomePageBase.class)
+public class WelcomePage extends WelcomePageBase implements IMobileUtils, IConstantUtils {
+
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/WelcomePage.java:16]: see above

File: src/main/java/koval/myfit/mobile/gui/pages/android/menuPages/BrowsePage.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import com.zebrunner.carina.utils.factory.DeviceType;
+import com.zebrunner.carina.utils.mobile.IMobileUtils;
+import koval.myfit.mobile.gui.pages.common.menuPages.BrowsePageBase;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = BrowsePageBase.class)
+public class BrowsePage extends BrowsePageBase implements IMobileUtils {
+    public BrowsePage(WebDriver driver) {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/menuPages/BrowsePage.java:15]: add space here

File: src/main/java/koval/myfit/mobile/gui/pages/android/menuPages/HomePage.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        super(driver);
+    }
+
+    @Override
+    public boolean isPageOpened() {
+        return activityChartView.isElementPresent(5);
+    }
+
+    @Override
+    public WelcomePageBase clickSignOutBtn() {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/menuPages/HomePage.java]: incorrect name of method. here you provide two different clicks on different locators

File: src/main/java/koval/myfit/mobile/gui/pages/ios/loginPages/AboutMePage.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import com.zebrunner.carina.utils.mobile.IMobileUtils;
+import koval.carinademo.mobile.gui.pages.service.interfaces.IConstantUtils;
+import koval.myfit.mobile.gui.pages.common.loginPages.AboutMePageBase;
+import koval.myfit.mobile.gui.pages.common.loginPages.TrackActivitiesPageBase;
+import koval.myfit.mobile.gui.pages.service.enums.PersonCharacteristics;
+import org.openqa.selenium.WebDriver;
+
+
+@DeviceType(pageType = DeviceType.Type.IOS_PHONE, parentClass = AboutMePageBase.class)
+public class AboutMePage extends AboutMePageBase implements IMobileUtils, IConstantUtils {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/ios/loginPages/AboutMePage.java]: see above

File: src/main/java/koval/myfit/mobile/gui/pages/ios/loginPages/TrackActivitiesPage.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import koval.carinademo.mobile.gui.pages.service.interfaces.IConstantUtils;
+import koval.myfit.mobile.gui.pages.common.loginPages.TrackActivitiesPageBase;
+import koval.myfit.mobile.gui.pages.common.menuPages.HomePageBase;
+import koval.myfit.mobile.gui.pages.service.enums.TrackYourActivitiesAnswer;
+import org.openqa.selenium.WebDriver;
+
+
+@DeviceType(pageType = DeviceType.Type.IOS_PHONE, parentClass = TrackActivitiesPageBase.class)
+public class TrackActivitiesPage extends TrackActivitiesPageBase implements IMobileUtils, IConstantUtils {
+
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/ios/loginPages/TrackActivitiesPage.java:16]: see above

File: src/main/java/koval/myfit/mobile/gui/pages/ios/loginPages/WelcomePage.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import com.zebrunner.carina.utils.mobile.IMobileUtils;
+import koval.carinademo.mobile.gui.pages.service.interfaces.IConstantUtils;
+import koval.myfit.mobile.gui.pages.common.loginPages.AboutMePageBase;
+import koval.myfit.mobile.gui.pages.common.loginPages.WelcomePageBase;
+import org.openqa.selenium.WebDriver;
+
+
+@DeviceType(pageType = DeviceType.Type.IOS_PHONE, parentClass = WelcomePageBase.class)
+public class WelcomePage extends WelcomePageBase implements IMobileUtils, IConstantUtils {
+    public WelcomePage(WebDriver driver) {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/ios/loginPages/WelcomePage.java:15]: see above and add space

File: src/main/java/koval/myfit/mobile/gui/pages/ios/menuPages/JournalPage.java [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.zebrunner.carina.utils.factory.DeviceType;
+import com.zebrunner.carina.utils.mobile.IMobileUtils;
+import koval.myfit.mobile.gui.pages.common.menuPages.JournalPageBase;
+import org.openqa.selenium.WebDriver;
+
+
+@DeviceType(pageType = DeviceType.Type.IOS_PHONE, parentClass = JournalPageBase.class)
+public class JournalPage extends JournalPageBase implements IMobileUtils {
+    public JournalPage(WebDriver driver) {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/ios/menuPages/JournalPage.java:12]: see above

File: src/main/java/koval/myfit/mobile/gui/pages/ios/menuPages/ProfilePage.java [hunk 12]
```diff
@@ -0,0 +1,13 @@
+package koval.myfit.mobile.gui.pages.ios.menuPages;
+
+import com.zebrunner.carina.utils.factory.DeviceType;
+import com.zebrunner.carina.utils.mobile.IMobileUtils;
+import koval.myfit.mobile.gui.pages.common.menuPages.ProfilePageBase;
+import org.openqa.selenium.WebDriver;
+
+@DeviceType(pageType = DeviceType.Type.IOS_PHONE, parentClass = ProfilePageBase.class)
+public class ProfilePage extends ProfilePageBase implements IMobileUtils {
+    public ProfilePage(WebDriver driver) {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/ios/menuPages/ProfilePage.java:10]: see above

File: src/main/java/koval/myfit/mobile/gui/pages/service/enums/PersonCharacteristics.java [hunk 13]
```diff
@@ -0,0 +1,19 @@
+package koval.myfit.mobile.gui.pages.service.enums;
+
+public enum PersonCharacteristics {
+    GENDER("gender"),
+    BIRTHDAY("birthday"),
+    WEIGHT("weight"),
+    HEIGHT("height");
+
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/service/enums/PersonCharacteristics.java:8]: it's good idea if you fave the same ids on  more than one pages

File: src/main/java/koval/myfit/mobile/gui/pages/service/AdbService.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
+package koval.myfit.mobile.gui.pages.service;
+
+import java.io.IOException;
+
+
+import static koval.myfit.mobile.gui.pages.service.interfaces.IConstantUtils.ADB_PATH;
+
+public class AdbService {
+
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/service/AdbService.java:12]: Create here enum AppPackage and try to implement

  private static AndroidService executor = AndroidService.getInstance();

    public void clearAppCache(AppPackage packageName) {
        LOGGER.info("Clear ['{}'] cache via ADB", packageName);
        String[] command = { "adb", "-s", getDevice().getUdid(), "shell", "pm", "clear", packageName.getPackageName() };
        getDevice().execute(command);
    }
    
        public void startApp(AppPackage packName) {
        LOGGER.info("Starting " + packName.toString());
        String cmd = String.format("shell monkey -p %s -c android.intent.category.LAUNCHER 1", packName.getPackageName());
        executor.executeAdbCommand(cmd);
        pause(TimeConstants.FIVE_SEC_TIMEOUT);
    }

File: src/main/java/koval/myfit/mobile/gui/pages/service/interfaces/IConstantUtils.java [hunk 15]
```diff
@@ -0,0 +1,7 @@
+package koval.myfit.mobile.gui.pages.service.interfaces;
+
+public interface IConstantUtils {
+
+    int TIMEOUT_FIVE = 5;
+    String ADB_PATH = "/Users/dianakoval/Library/Android/sdk/platform-tools/adb";
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/service/interfaces/IConstantUtils.java]: its won't work for teammates

**Review by YanaDorosh — COMMENTED** [review]
you did a great job!!! 👏 awesome

added a few minor comments

File: src/main/java/koval/myfit/mobile/gui/pages/service/AdbService.java (hunk 14 above)
Comment (DianaKovall) [reply src/main/java/koval/myfit/mobile/gui/pages/service/AdbService.java:12]: I had trouble with this task, but I solved it! 
1. Error [Cannot run program "adb": error=2, No such file or directory] -> write ANDROID_HOME path to .zshrc file worked for me
2. getDevice() return null -> getDevice(getDriver)

File: src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/AboutMePage.java (hunk 2 above)
Comment (DianaKovall) [reply src/main/java/koval/myfit/mobile/gui/pages/android/loginPages/AboutMePage.java:53]: For now it/s only one(AboutMe Page)

File: src/main/java/koval/myfit/mobile/gui/pages/service/MyAbstractPage.java [hunk 16]
```diff
@@ -0,0 +1,12 @@
+package koval.myfit.mobile.gui.pages.service;
+
+import com.qaprosoft.carina.core.gui.AbstractPage;
+import com.zebrunner.carina.utils.mobile.IMobileUtils;
+import koval.myfit.mobile.gui.pages.service.interfaces.IConstantUtils;
+import org.openqa.selenium.WebDriver;
+
+public abstract class MyAbstractPage extends AbstractPage implements IConstantUtils, IMobileUtils {
+    public MyAbstractPage(WebDriver driver) {
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/service/MyAbstractPage.java:10]: space

File: src/test/java/koval/myfit/GoogleFitTest.java [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.lang.invoke.MethodHandles;
+
+
+public class GoogleFitTest extends LoginTest implements IAbstractTest, IMobileUtils {
+
```
Comment (YanaDorosh) [inline src/test/java/koval/myfit/GoogleFitTest.java:17]: do really need it here?

**Review by YanaDorosh — APPROVED** [review]

File: src/main/java/koval/myfit/mobile/gui/pages/android/menuPages/HomePage.java [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Override
+    public HomePageBase clickProfileImage() {
+        profileImage.click();
+        return initPage(getDriver(), HomePageBase.class);
+    }
+
+    @Override
+    public WelcomePageBase clickSignOutBtn() {
+        clickProfileImage();
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/menuPages/HomePage.java]: incorrect name of method clickSignOutBtn for you current implementation the better name is signOut()
you don't need to use here clickProfileImage() use profileImage.click(); avoid usage method inside another method if it possible

File: src/main/java/koval/myfit/mobile/gui/pages/android/menuPages/HomePage.java [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+    @FindBy(id = "com.google.android.apps.fitness:id/og_text_card_title")
+    private ExtendedWebElement signOutButton;
+
+    public HomePage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    public boolean isPageOpened() {
+        return activityChartView.isElementPresent(5);
```
Comment (YanaDorosh) [inline src/main/java/koval/myfit/mobile/gui/pages/android/menuPages/HomePage.java]: use understandable constants for timeout

**Pull request merged**